tree/ntuple/v7/inc/ROOT/RNTupleProcessor.hxx

```diff
@@ -273,6 +292,49 @@ private:
    RNTupleChainProcessor(const std::vector<RNTupleSourceSpec> &ntuples, std::unique_ptr<RNTupleModel> model = nullptr);
 };
 
+// clang-format off
+/**
+\class ROOT::Experimental::RNTupleJoinProcessor
+\ingroup NTuple
+\brief Processor specializiation for horizontally concatenated RNTuples (joins).
+*/
+// clang-format on
+class RNTupleJoinProcessor : public RNTupleProcessor {
+   friend class RNTupleProcessor;
+
+private:
+   std::unique_ptr<RNTupleModel> fJoinModel;
+   std::unordered_map<std::string, std::unique_ptr<Internal::RPageSource>> fAuxiliaryPageSources;
+
+   bool fUseIndex = false;
+   std::vector<std::string> fJoinFieldNames;
+   std::unordered_map<std::string, std::unique_ptr<Internal::RNTupleIndex>> fJoinIndices;
+
+   NTupleSize_t Advance() final;
+   void LoadEntry() final;
+
+   /////////////////////////////////////////////////////////////////////////////
+   /// \brief Constructs a new RNTupleJoinProcessor.
+   ///
+   /// \param[in] ntuples The source specification (name and storage location) of the primary RNTuple.
+   /// \param[in] model The model that specifies which fields should be read by the processor. The pointer returned by
+   /// RNTupleModel::MakeField can be used to access a field's value during the processor iteration. When no model is
+   /// specified, it is created from the descriptor of the first RNTuple specified in `ntuples`.
+   RNTupleJoinProcessor(const RNTupleSourceSpec &mainNTuple, const std::vector<std::string> &joinFields,
+                        std::unique_ptr<RNTupleModel> model = nullptr);
+
+   void AddAuxiliary(const RNTupleSourceSpec &auxNTuple, std::unique_ptr<RNTupleModel> model = nullptr);
+   void ConnectFields();
+
+public:
+   ~RNTupleJoinProcessor()
```

Comment:
Missing other rule of five methods. Also, since this class derives from an abstract class, the destructor of the base should be virtual and this destructor should override